src/classes/SRVC_Contact_PreventDeletion_TEST.cls

```diff
@@ -0,0 +1,310 @@
+
+        Application__c applicant = UTIL_UnitTestData_TEST.getApplication(acc.Id, contacts[0].Id, term.Id);
+        insert applicant;
+
+        Application__c applicantPreparer = UTIL_UnitTestData_TEST.getApplication(acc.Id, null, term.Id);
+        applicantPreparer.Preparer__c = contacts[0].Id;
+        insert applicantPreparer;
+
+        Attendance_Event__c attendance = UTIL_UnitTestData_TEST.getAttendanceEvent(contacts[0].Id, null, null, null);
+        insert attendance;
+
+        Id credentialRecTypeId = UTIL_Describe_API.getCredentialAttributeRecType();
+        Attribute__c credentialAttribute = UTIL_UnitTestData_TEST.getAttribute('TestCred2', contacts[0].Id,  credentialRecTypeId);
+        insert credentialAttribute;
+
+        Case caseRecord = UTIL_UnitTestData_TEST.getCase(contacts[0].Id, 'School Violation', 'New');
+        insert caseRecord;
+
+        Language__c language = UTIL_UnitTestData_TEST.createLanguage('Spanish');
+
+        Contact_Language__c contactLanguage = UTIL_UnitTestData_TEST.createContactLanguage(language.Id, contacts[0].Id, false);
+
+        Course_Offering__c courseOffering = UTIL_UnitTestData_TEST.createCourseOffering(course.Id, term.Id);
+        courseOffering.Faculty__c = contacts[0].Id;
+        update courseOffering;
+
+        Course_Enrollment__c courseCxn = UTIL_UnitTestData_TEST.getCourseConnection(contacts[0].Id, courseOffering.Id);
+        insert courseCxn;
+
+        Term_Grade__c termGrade = UTIL_UnitTestData_TEST.getTermGrade(contacts[0].Id,  courseOffering.Id, courseCxn.Id);
+        insert termGrade;
+
+        Program_Enrollment__c programEnrollment = UTIL_UnitTestData_TEST.getProgramEnrollment(acc.Id);
+        programEnrollment.Contact__c = contacts[0].Id;
+        insert programEnrollment;
+
+        Test__c testRecord = UTIL_UnitTestData_TEST.getTest('SAT', System.today(), contacts[0].Id);
+        insert testRecord;
+
+        Behavior_Involvement__c behInvolvement = UTIL_UnitTestData_TEST.getBehaviorInvolvementRecord(contacts[0].Id, caseRecord.Id, 'Reporter');
+        insert behInvolvement;
+
+    }
+
+    // Tests getInstance()
+    @isTest
+    static void testGetInstance() {
+
+        SRVC_Contact_PreventDeletion srvcConPreventDeleteInstance = SRVC_Contact_PreventDeletion.getInstance();
+        System.assertEquals(SRVC_Contact_PreventDeletion.instance, srvcConPreventDeleteInstance);
+        System.assertNotEquals(NULL, srvcConPreventDeleteInstance);
+
+        srvcConPreventDeleteInstance = SRVC_Contact_PreventDeletion.getInstance();
+        System.assertEquals(SRVC_Contact_PreventDeletion.instance, srvcConPreventDeleteInstance);
+    }
+
+    // Tests Prevent Deletion for Contacts with STUB_AlwaysHasChildRelationships
+    @isTest
+    static void testPreventDeletionForContactsWithChildren() {
+
+        SRVC_Contact_PreventDeletion.instance = new STUB_AlwaysHasChildRelationships();
+
+        Map<Id,Contact> contactById = new Map<Id,Contact>();
```

Comment:
Please add contacts to this map. Note that we do not need to insert them thanks to the magic of stubbing.

Reply:
@MatthewBlanski  shouldn't we be inserting the Contact here (with no Children)? Otherwise, it would not enter the for loop that would call the overriden method?

Reply:
@skosgi1 You are accurate here. You may want to create another stub that overrides getContactsAndChildRelationships and contactHasChildRecords to accomplish this. The stub could cause getContactsAndChildRelationships to return a list of contacts created by new Contact(Id = idInSet, LastName = idInSet);

Reply:
@skosgi1 Is this item resolved?

Reply:
It's resolved @spinkelman!